ZionEscape/Scene.h

```diff
@@ -0,0 +1,155 @@
+
+      else if (this->up && this->down && this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\G.png");
+
+      else if (!this->up && !this->down && !this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\L.png");
+
+      else if (!this->up && !this->down && this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\RL.png");
+
+      else if (!this->up && !this->down && this->right && !this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\R.png");
+
+      else if (this->up && !this->down && !this->right && !this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\T.png");
+
+      else if (this->up && this->down && !this->right && !this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\TD.png");
+
+      else if (this->up && !this->down && !this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\TL.png");
+
+      else if (this->up && !this->down && this->right && !this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\TR.png");
+
+      else if (!this->up && this->down && !this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\DL.png");
+
+      else if (!this->up && this->down && this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\DRL.png");
+
+      else if (this->up && this->down && !this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\TDL.png");
+
+      else if (this->up && this->down && this->right && !this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\TDR.png");
+
+      else if (this->up && !this->down && this->right && this->left)
+        this->image = gcnew Bitmap("assets\\sprites\\colliders\\TRL.png");
+  }
+
+  void Draw(Graphics^ g) {
+    g->DrawImage(this->image, this->drawingArea);
+  }
+
+  void CreateSpawner(Point pos) {
+    Point location;
+    Size size = Size(this->image->Width, this->image->Height);
+    if (this->up) {
+      location = Point(pos.X, pos.Y - size.Height);
+      this->spawners->Add(gcnew SceneSpawner(ParentDirection::Down, Rectangle(location, size)));
+    }
+    if (this->down) {
+      location = Point(pos.X, pos.Y + size.Height);
+      this->spawners->Add(gcnew SceneSpawner(ParentDirection::Up, Rectangle(location, size)));
+    }
+    if (this->right) {
+      location = Point(pos.X + size.Width, pos.Y);
+      this->spawners->Add(gcnew SceneSpawner(ParentDirection::Left, Rectangle(location, size)));
+    }
+    if (this->left) {
+      location = Point(pos.X - size.Width, pos.Y);
+      this->spawners->Add(gcnew SceneSpawner(ParentDirection::Right, Rectangle(location, size)));
+    }
+  }
+
+  void DeleteSpawner(short n) {
+    this->spawners->Remove(this->spawners[n]);
+  }
```

Comment:
Por si acaso, estaría mejor tener alguna validación.

```suggestion
  void DeleteSpawner(short n) {
    if (n >= 0 && n < this->spawners->Count)
      this->spawners->Remove(this->spawners[n]);
  }
```

Reply:
Me parece que la verificación no es necesaria. Ya que el método recibe, en un momento determinado, un parámetro en específico (*en este caso `n`*), por lo que obligatoriamente ese `spawner` va a existir.